src/sources/util/net/tcp/mod.rs

```diff
@@ -351,16 +351,13 @@ async fn handle_stream<T>(
                             for event in &mut events {
                                 let log = event.as_mut_log();
 
-                                match log_namespace {
-                                    LogNamespace::Vector => {
-                                        log.insert(metadata_path!(source_name, "tls_client_metadata"), metadata.clone());
-                                    }
-                                    LogNamespace::Legacy => {
-                                        if let Some(tls_client_metadata_key) = &tls_client_metadata_key {
-                                            log.insert(&tls_client_metadata_key[..], metadata.clone());
```

Comment:
:thinking:  isn't this change, modifying the shape of the legacy namespace use case?

Reply:
I don't think so, either case should Overwrite into whatever String/Path `tls_client_metadata_key` contains. The `insert` just used a [string slice](https://doc.rust-lang.org/book/ch04-03-slices.html#string-slices) because:

```text
the trait `TargetPath<'_>` is not implemented for `&std::string::String`
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d572d751263fb404f07350233679c497

Where/how do you think it changes?

Reply:
Ah, my bad, I just didn't look at it closely enough. I see now.